--- setup.py.orig
+++ setup.py
@@ -1,3 +1,4 @@
+import sys
 import os
 import io
 import re
@@ -385,6 +386,8 @@ def get_extension_modules():
         "src/libImaging/Resample.c",
     ]
 
+    if sys.platform == "android":
+        extra_compile_args.append("-fPIC"); extra_link_args.append("-fPIC")
     if not CFLAGS:
         # TODO: remove when all warnings are fixed
         CFLAGS = ["-Wno-unused-value", "-Wno-empty-body"]
@@ -455,6 +458,8 @@ def get_extension_modules():
             "-Wno-format-truncation",
         ]
     
+    if sys.platform == "android":
+        libraries = []
     _add_imaging_libs(libs, include_dirs, library_dirs, libraries)