--- setup.py.orig	2024-01-01 00:00:00.000000000 +0000
+++ setup.py	2024-01-01 00:00:00.000000000 +0000
@@ -1,4 +1,5 @@
 import os
+import sys
 import io
 import re
 import sysconfig
@@ -390,6 +391,8 @@ def get_extension_modules():
         "src/libImaging/Quant.c",
         "src/libImaging/Resample.c",
     ]
+    if sys.platform == "android":
+        extra_compile_args.append("-fPIC"); extra_link_args.append("-fPIC")
     
     if not CFLAGS:
         # TODO: remove when all warnings are fixed
@@ -460,6 +463,7 @@ def get_extension_modules():
             "-Wno-format-truncation",
         ]
     
+    if sys.platform == "android": libraries = []
     
     _add_imaging_libs(libs, include_dirs, library_dirs, libraries)